app/src/main/java/org/sil/hearthis/MainActivity.java

```diff
@@ -1,76 +1,220 @@
 
 	@Override
-	protected void onCreate(Bundle savedInstanceState) {
-		super.onCreate(savedInstanceState);
+	protected void onCreate(Bundle state)
+	{
+		super.onCreate(state);
+
         Graphite.loadGraphite();
 		setContentView(R.layout.activity_main);
+
 		// Todo: scan org.sil.hearthis/files for folders containing info.txt and open first
-		// Todo: remember last project
-		// Todo: if no real project available use SampleScriptProvider.
-		RealScriptProvider scripture = new RealScriptProvider(getExternalFilesDir(null) + "/" + "Dhh");
+
+		IScriptProvider scripture;
+		try
+		{
+			scripture = new RealScriptProvider(getExternalFilesDir(null) + "/" + "Dhh");
+		}
+		catch (Exception ex)
+		{
+			scripture = new SampleScriptProvider();
+		}
 		Project project = new Project("Sample", scripture);
-		setProject(project);
+		try
+		{
+			setProject(project);
+		}
+		catch (Exception ex)
+		{
+			ex.printStackTrace();
+		}
 	}
-	
-	public void setProject(Project project) {
-		LayoutInflater inflater = (LayoutInflater) getSystemService(Context.LAYOUT_INFLATER_SERVICE);
-		ViewGroup bookFlow = (ViewGroup) findViewById(R.id.booksFlow);
-		for (BookInfo book : project.Books) {
-			int resid = R.layout.book_button;
-			if (book.BookNumber == 39) {
-				// Matthew: start new line
-				resid = R.layout.newline_book_button;
-			}
-			
-			// This next line is rather non-obvious. We must pass the bookFlow to the inflator
-			// so that it can be used to create the Layout for the button: a custom layout which implements
-			// the extra properties that the FlowLayout recognizes for its children like newline.
-			// But, when we pass a parent, without the third argument inflate adds the button to the
-			// flowLayout itself and returns the parent. That leaves us without an easy way to get the
-			// new button, on which we want to set other properties. So we pass false (do not add
-			// to parent) and thus get the button itself back from inflate. Then of course we must
-			// add it to the parent ourselves.
-			BookButton bookButton = (BookButton)inflater.inflate(resid, bookFlow, false);
-			bookButton.Model = book;
-			bookButton.setOnClickListener(bookButtonListener);
-			bookButton.setTag(book);
-			bookFlow.addView(bookButton);
+
+	@Override
+	public boolean onOptionsItemSelected(MenuItem item)
+	{
+		switch (item.getItemId())
+		{
+		case R.id.open:
+			Toast.makeText(this, "Open", Toast.LENGTH_LONG).show();
+			return true;
+		case R.id.publish:
+			Toast.makeText(this, "Publish", Toast.LENGTH_LONG).show();
+			return true;
 		}
+		return super.onOptionsItemSelected(item);
 	}
-	
-	public android.view.View.OnClickListener bookButtonListener = new android.view.View.OnClickListener() {
-		
+
+	@Override
+	public boolean onCreateOptionsMenu(Menu menu)
+	{
+		mainMenu = menu;
+		MenuInflater inflater = getMenuInflater();
+		inflater.inflate(R.menu.menu_main, menu);
+		return super.onCreateOptionsMenu(menu);
+	}
+
+	public void setProject(Project project)
```

Comment:
Again, moving existing methods makes it hard to see what else you changed. If you want to re-arrange a file, it's best to make a separate PR.